src/journal/journalctl-misc.c

```diff
@@ -120,21 +121,21 @@ static int show_log_ids(const LogId *ids, size_t n_ids, const char *name) {
         (void) table_set_sort(table, (size_t) 0);
         (void) table_set_reverse(table, 0, arg_reverse);
 
-        for (int i = 0; i < (int) n_ids; i++) {
-                int index;
+        for (size_t i = 0; i < n_ids; i++) {
+                int64_t index;
 
                 if (arg_lines_needs_seek_end())
                         /* With --lines=N, we only know the negative index, and the older ID is located earlier. */
-                        index = -i;
+                        index = - (int64_t) i;
                 else if (arg_lines >= 0)
                         /* With --lines=+N, we only know the positive index, and the newer ID is located earlier. */
                         index = i + 1;
                 else
                         /* Otherwise, show negative index. Note, in this case, newer ID is located earlier. */
-                        index = i + 1 - (int) n_ids;
+                        index = (int64_t) (i + 1) - (int64_t) n_ids;
```

Comment:
If you touch this, please correct the comment here (https://github.com/systemd/systemd/pull/32491#discussion_r1596705037). We actually always show older ids earlier, no matter which mode of `--lines=` is used.

If `advance_older` is false, `boots` is ordered from older -> newer, and index is always increasing, no matter positive or negative.

Reply:
Sorry, I cannot follow... I explicitly added all combinations `-n` is positive and negative, and `--reverse` is enabled and disabled. And the results look expected.